website/blog/2023-08-18-deploy-run-ferretdb-leafcloud.md

```diff
@@ -0,0 +1,491 @@
+```
+
+Note that you can get the `ferretuser` password by running:
+
+```sh
+kubectl get secret ferretdb-secret -n postgres-operator -o jsonpath='{.data.password}' | base64 --decode
+```
+
+And that's it!.
+You've connected to FerretDB.
+
+## Basic examples on FerretDB
+
+Let's run a few basic examples using FerretDB:
+
+Insert documents into the database:
+
+```js
+db.testing.insertMany([
+  { a: 23, b: 'b', c: [1, 5], d: { a: 1 } },
+  { a: 1, b: 34, c: '1', d: [3, 5] }
+])
+```
+
+Now let's read all these documents and see what we get.
+
+```js
+db.testing.find()[
+  ({
+    _id: ObjectId('64ca02e119e6b74d10806107'),
+    a: 23,
+    b: 'b',
+    c: [1, 5],
+    d: { a: 1 }
+  },
+  {
+    _id: ObjectId('64ca02e119e6b74d10806108'),
+    a: 1,
+    b: 34,
+    c: '1',
+    d: [3, 5]
+  })
+]
+```
+
+### How FerretDB stores data in PostgreSQL
+
+We can take a look our data in PostgreSQL to see how the FerretDB conversion works out.
+
+In another terminal, let's establish a connection to your PostgreSQL cluster.
+
+```sh
+kubectl exec -it hippo-instance1-mrpt-0 -n postgres-operator -- psql -U postgres
+```
+
+Let's switch to the new `ferretdb` database context:
+
+```text
+postgres=# \c ferretdb
+You are now connected to database "ferretdb" as user "postgres".
+```
+
+Finally we're going to set the `search_path` to ferretdb:
+
+```sql
+set search_path to ferretdb;
+```
+
+```text
+ferretdb=# \dt
+                     List of relations
+  Schema  |            Name             | Type  |  Owner
+----------+-----------------------------+-------+----------
+ ferretdb | _ferretdb_database_metadata | table | ferretdb
+ ferretdb | testing_eb5f499b            | table | ferretdb
+(2 rows)
+
+ferretdb=# table testing_eb5f499b;
+                                                                                                                                                                _jsonb
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ {"a": 23, "b": "b", "c": [1, 5], "d": {"a": 1}, "$s": {"p": {"a": {"t": "int"}, "b": {"t": "string"}, "c": {"i": [{"t": "int"}, {"t": "int"}], "t": "array"}, "d": {"t": "object", "$s": {"p": {"a": {"t": "int"}}, "$k": ["a"]}}, "_id": {"t": "objectId"}}, "$k": ["_id", "a", "b", "c", "d"]}, "_id": "64ca02e119e6b74d10806107"}
+ {"a": 1, "b": 34, "c": "1", "d": [3, 5], "$s": {"p": {"a": {"t": "int"}, "b": {"t": "int"}, "c": {"t": "string"}, "d": {"i": [{"t": "int"}, {"t": "int"}], "t": "array"}, "_id": {"t": "objectId"}}, "$k": ["_id", "a", "b", "c", "d"]}, "_id": "64ca02e119e6b74d10806108"}
+(2 rows)
+~
+
+```
+
+## Conclusion
+
+FerretDB offers you the chance to leverage the truly open-source replacement for MongoDB.
```

Comment:
FerretDB offers a truly open source replacement for MongoDB. It is compatible with MongoDB, which means you can keep using your existing tools and frameworks.

Reply:
I'd use the above sentence instead of the existing one